drivers/sensor/bq274xx/bq274xx.h

```diff
@@ -0,0 +1,209 @@
+#define BQ274XX_COMMAND_SOC			0x1C		/* StateOfCharge() */
+#define BQ274XX_COMMAND_INT_TEMP		0x1E		/* InternalTemperature() */
+#define BQ274XX_COMMAND_SOH			0x20		/* StateOfHealth() */
+#define BQ274XX_COMMAND_REM_CAP_UNFL		0x28		/* RemainingCapacityUnfiltered() */
+#define BQ274XX_COMMAND_REM_CAP_FIL		0x2A		/* RemainingCapacityFiltered() */
+#define BQ274XX_COMMAND_FULL_CAP_UNFL		0x2C		/* FullChargeCapacityUnfiltered() */
+#define BQ274XX_COMMAND_FULL_CAP_FIL		0x2E		/* FullChargeCapacityFiltered() */
+#define BQ274XX_COMMAND_SOC_UNFL		0x30		/* StateOfChargeUnfiltered() */
+
+
+/*** Control Sub-Commands ***/
+#define BQ274XX_CONTROL_STATUS			0x0000
+#define BQ274XX_CONTROL_DEVICE_TYPE		0x0001
+#define BQ274XX_CONTROL_FW_VERSION		0x0002
+#define BQ274XX_CONTROL_DM_CODE			0x0004
+#define BQ274XX_CONTROL_PREV_MACWRITE		0x0007
+#define BQ274XX_CONTROL_CHEM_ID			0x0008
+#define BQ274XX_CONTROL_BAT_INSERT		0x000C
+#define BQ274XX_CONTROL_BAT_REMOVE		0x000D
+#define BQ274XX_CONTROL_SET_HIBERNATE		0x0011
+#define BQ274XX_CONTROL_CLEAR_HIBERNATE		0x0012
+#define BQ274XX_CONTROL_SET_CFGUPDATE		0x0013
+#define BQ274XX_CONTROL_SHUTDOWN_ENABLE		0x001B
+#define BQ274XX_CONTROL_SHUTDOWN		0x001C
+#define BQ274XX_CONTROL_SEALED			0x0020
+#define BQ274XX_CONTROL_PULSE_SOC_INT		0x0023
+#define BQ274XX_CONTROL_RESET			0x0041
+#define BQ274XX_CONTROL_SOFT_RESET		0x0042
+#define BQ274XX_CONTROL_EXIT_CFGUPDATE		0x0043
+#define BQ274XX_CONTROL_EXIT_RESIM		0x0044
+
+/*** Extended Data Commands ***/
+#define BQ274XX_EXTENDED_OPCONFIG			0x3A		/* OpConfig() */
+#define BQ274XX_EXTENDED_CAPACITY			0x3C		/* DesignCapacity() */
+#define BQ274XX_EXTENDED_DATA_CLASS			0x3E		/* DataClass() */
+#define BQ274XX_EXTENDED_DATA_BLOCK			0x3F		/* DataBlock() */
+#define BQ274XX_EXTENDED_BLOCKDATA_START		0x40		/* BlockData_start() */
+#define BQ274XX_EXTENDED_BLOCKDATA_END			0x5F		/* BlockData_end() */
+#define BQ274XX_EXTENDED_CHECKSUM			0x60		/* BlockDataCheckSum() */
+#define BQ274XX_EXTENDED_DATA_CONTROL			0x61		/* BlockDataControl() */
+#define BQ274XX_EXTENDED_BLOCKDATA_DESIGN_CAP_HIGH	0x4A		/* BlockData */
+#define BQ274XX_EXTENDED_BLOCKDATA_DESIGN_CAP_LOW	0x4B
+#define BQ274XX_EXTENDED_BLOCKDATA_DESIGN_ENR_HIGH	0x4C
+#define BQ274XX_EXTENDED_BLOCKDATA_DESIGN_ENR_LOW	0x4D
+#define BQ274XX_EXTENDED_BLOCKDATA_TERMINATE_VOLT_HIGH	0x50
+#define BQ274XX_EXTENDED_BLOCKDATA_TERMINATE_VOLT_LOW	0x51
+#define BQ274XX_EXTENDED_BLOCKDATA_TAPERRATE_HIGH	0x5B
+#define BQ274XX_EXTENDED_BLOCKDATA_TAPERRATE_LOW	0x5C
+
+#define BQ274XX_DELAY			1000
+
+struct bq274xx_data {
+	struct device *i2c;
+	u16_t voltage;
+	s16_t avg_current;
+	s16_t stdby_current;
+	s16_t max_load_current;
+	s16_t avg_power;
+	u16_t state_of_charge;
+	s16_t state_of_health;
+	u16_t internal_temperature;
+	u16_t full_charge_capacity;
+	u16_t remaining_charge_capacity;
+	u16_t nom_avail_capacity;
+	u16_t full_avail_capacity;
+};
+
+
+
+int bq274xx_command_reg_read(struct bq274xx_data *bq274xx, u8_t reg_addr,
```

Comment:
why do you have function implementations in the header file? please move them to .c

Reply:
Moved to .c, thanks.